File: RepoTransBench/target_projects/Java/C#/JamesZBL_java_design_patterns/tests/original/WriterIntegrationTest.cs
using System;
using System.IO;
using Xunit;

namespace Tests.Original
{
    public class WriterIntegrationTest
    {
        [Fact]
        public void TestWriterSentences()
        {
            var output = new StringWriter();
            var oldOut = Console.Out;
            Console.SetOut(output);
            // You must implement the Writer class in src/MainProject mimicking Java's logic
            new Writer().SentenceByChinese().Print();
            new Writer().SentenceByEnglish().Print();
            Console.SetOut(oldOut);
            string result = output.ToString();
            Assert.Contains(".", result);
        }
    }
}

File: RepoTransBench/source_projects/Java/JamesZBL_java_design_patterns/composite/src/test/java/me/zbl/composite/WriterIntegrationTest.java
package me.zbl.composite;

import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

class WriterIntegrationTest {
    @Test
    void testWriterSentences() {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        PrintStream old = System.out;
        System.setOut(new PrintStream(baos));
        // test sentenceByChinese and sentenceByEnglish print output
        new Writer().sentenceByChinese().print();
        new Writer().sentenceByEnglish().print();
        System.setOut(old);
        String result = baos.toString();
        // Ensure at least the result contains a Chinese-like character and a . at end
        assert(result.contains("."));
    }
}